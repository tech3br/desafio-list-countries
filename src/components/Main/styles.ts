import styled from 'styled-components'

export const Wrapper = styled.div`

`
export const Title = styled.div`

`
export const PanelContent = styled.div`

`
