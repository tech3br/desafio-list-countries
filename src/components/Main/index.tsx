import * as React from "react";
import * as S from "./styles";

const Main = () => {
  return <S.Wrapper>
    <S.Title></S.Title>
  </S.Wrapper>;
};

export default Main;
