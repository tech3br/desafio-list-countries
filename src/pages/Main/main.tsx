import {
  Box,
  Button,
  Card,
  CardActions,
  CardContent,
  CardMedia,
  Typography
} from "@material-ui/core/";
import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import ModalComponent from "../../components/Modal/Modal";
import { GET_COUNTRIES } from "../../graphql/queries/get_countries";
import { Country } from "../../redux/store/countries/types";
import { useStyles } from "./styles";

//definido o tipo do dado em erros
interface Errors {
  text: string;
}

interface Variables {
  first: number;
  offset: number;
  limit: number;
}

interface ModalProps {
  value: boolean;
}

interface Name {
  value: string;
}

//definindo o componente como funcional
const Main = () => {
  //guardando os estilos do material-ui numa constante styles
  const styles = useStyles();
  //definindo um dispatch para editar um pais
  //definindo estado para guardar os paises
  const [state, setState] = useState<Country[]>();
  //definindo estado para guardar um possivel erro
  const [errors, setErrors] = useState<Errors>();
  //definindo o total
  // const [total, setTotal] = useState<>();
  const [isOpen, setIsOpen] = useState<ModalProps>({ value: false });
  const [name, setName] = useState<Name>({ value: "ola" });

  // dispatch(
  //   getCountry({})
  // );

  function handleClose() {
    setIsOpen({ value: false });
  }

  function handleOpen() {
    setIsOpen({ value: true });
    setName({ value: "ola mundo" });
  }

  function handleEdit() {
    return setIsOpen({ value: true });
  }

  //usando useEffect para trazer os dados assim que o componente foi montado
  useEffect(() => {
    GET_COUNTRIES.then((result) => setState(result.data.Country)).catch((err) =>
      setErrors(err)
    );
    console.log(state);
  }, []);

  return (
    <Box component="section" className={styles.container}>
      <Box component="header" className={styles.header}>
        Desafio Tecnico Softplan - Daniel Filgueira da Silva
      </Box>
      <Box component="section" className={styles.contentBox}>
        <Box component="header" className={styles.titleContentBox}>
          Paises
        </Box>
        {!errors ? (
          <Box component="section" className={styles.cards}>
            {state &&
              state.map((item) => (
                <>
                  <Card key={item?._id} className={styles.card}>
                    <CardMedia
                      className={styles.flag}
                      component="img"
                      alt="Contemplative Reptile"
                      image={item?.flag.svgFile}
                      title="Contemplative Reptile"
                    />
                    <CardContent className={styles.infosCard}>
                      <Typography
                        variant="body2"
                        color="textPrimary"
                        component="span"
                      >
                        <span style={{ fontWeight: 900 }}>Nome: </span>
                        {item?.name}
                      </Typography>
                      <Typography
                        variant="body2"
                        color="textPrimary"
                        component="span"
                      >
                        <span style={{ fontWeight: 900 }}>Capital: </span>
                        {item?.capital}
                      </Typography>
                    </CardContent>
                    <CardActions className={styles.buttonsCard}>
                      <Link className={styles.link} to="/details">
                        <Button className={styles.button}>Editar</Button>
                      </Link>
                      <Button className={styles.button} onClick={handleOpen}>
                        Detalhes
                      </Button>
                    </CardActions>
                  </Card>
                </>
              ))}
          </Box>
        ) : (
          <Box component="p" className={styles.error}>
            <p>
              Desculpe, algo deu errado ao carregar os dados! Erro:{" "}
              {errors.text}
            </p>
          </Box>
        )}
      </Box>
      {isOpen.value === true ? (
        <ModalComponent
          open={isOpen.value}
          textEdit={"editar"}
          textClose={"fechar"}
          handleClose={handleClose}
          handleEdit={handleEdit}
          titleModal={name.value}
          contentModal={"teste"}
        />
      ) : null}
    </Box>
  );
};

export default Main;
