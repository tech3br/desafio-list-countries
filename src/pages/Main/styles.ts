import { makeStyles } from "@material-ui/core/styles";
import { cor } from "../../global/colors.json";

export const useStyles = makeStyles({
  container: {
    background: cor.branca,
    color: "white",
    width: "100%",
    height: "100%",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "space-between",
  },

  header: {
    background: cor.tom5,
    color: "white",
    width: "100%",
    height: "40px",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
  },

  contentBox: {
    background: cor.cinza,
    width: "90%",
    height: "100%",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "flex-start",
    borderRadius: "10px",
    marginTop: "10px",    },

  titleContentBox: {
    background: cor.tom5,
    width: "100%",
    height: "40px",
    borderRadius: "10px",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
  },

  cards: {
    background: cor.branca,
    margin: "10px",
    borderRadius: "10px",
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    flexWrap: "wrap",
  },

  card: {
    height: "auto",
    marginTop: "20px",
    marginRight: "20px",
    boxShadow: "1px 1px 4px rgba(0,0,0,0.3)",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    width: "232px",
  },

  infosCard: {
    display: "flex",
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "flex-start",
  },

  buttonsCard: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
  },

  button: {
    textAlign: "center",
    color: "#fff",
    backgroundColor: "#5F6FDC",
    width: "200px",
    height: "40px",
    textTransform: "none",
    '&:hover': {
      background: "#000",
      color: "#fff"
    },
    textDecoration: "none", 
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    borderRadius: "10px",
  },

  link: {
    textDecoration: "none",    
  },

  error: {
    color: "black",
    fontSize: "18px",
  },

  flag: {
    width: "240px",
    height: "140px",    
  }
});
