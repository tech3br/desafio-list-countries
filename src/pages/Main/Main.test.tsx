import { render } from "@testing-library/react";
import React from "react";
import Main from "./main";

describe("Testando o componente Main", () => {
  it("should render correctly", () => {
    const wrapper = render(<Main />);
    expect(wrapper).toMatchSnapshot();
  });
});

test("o header foi renderizado?", () => {
  const { getByText } = render(<Main />);
  const x = getByText("Desafio Tecnico Softplan - Daniel Filgueira da Silva");
  expect(x).toEqual(x);
});
