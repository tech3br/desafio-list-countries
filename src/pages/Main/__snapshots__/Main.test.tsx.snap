// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando o componente Main should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="MuiBox-root MuiBox-root-13 makeStyles-container-1"
      >
        <header
          class="MuiBox-root MuiBox-root-14 makeStyles-header-2"
        >
          Desafio Tecnico Softplan - Daniel Filgueira da Silva
        </header>
        <section
          class="MuiBox-root MuiBox-root-15 makeStyles-contentBox-3"
        >
          <header
            class="MuiBox-root MuiBox-root-16 makeStyles-titleContentBox-4"
          >
            Paises
          </header>
          <section
            class="MuiBox-root MuiBox-root-17 makeStyles-cards-5"
          />
          <form
            class="search"
          >
            <input
              type="text"
              value="teste"
            />
            <input
              type="submit"
              value="Digite"
            />
          </form>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="MuiBox-root MuiBox-root-13 makeStyles-container-1"
    >
      <header
        class="MuiBox-root MuiBox-root-14 makeStyles-header-2"
      >
        Desafio Tecnico Softplan - Daniel Filgueira da Silva
      </header>
      <section
        class="MuiBox-root MuiBox-root-15 makeStyles-contentBox-3"
      >
        <header
          class="MuiBox-root MuiBox-root-16 makeStyles-titleContentBox-4"
        >
          Paises
        </header>
        <section
          class="MuiBox-root MuiBox-root-17 makeStyles-cards-5"
        />
        <form
          class="search"
        >
          <input
            type="text"
            value="teste"
          />
          <input
            type="submit"
            value="Digite"
          />
        </form>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
